### PR TITLE
Refactor GetDistributedGeometricRandomComponent function to DistributedGeometricNoiser class.

### PR DESCRIPTION
**Pull request by iverson52000**
For the changes in any-sketch PR https://github.com/world-federation-of-advertisers/any-sketch/pull/20

- - -
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)

### CODE REVIEW

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: all files reviewed, 5 unresolved discussions (waiting on @iverson52000, @kungfucraig, @uakyol, and @wangyaopw)

___
*[`.clang-format` line 1 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhU_B13iT-m8bcrnOU:-NNhU_B13iT-m8bcrnOV:bxzgsb0) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/.clang-format#L1)):*
> ```YAML
> Language: Cpp
> ```

We use `--style=Google` for more than just C++, e.g. protobuf. See https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/main/docs/code-style.md#other-languages

Note that it looks like this PR incorrectly reformats some code in a way that does not match this style.



_Code quote:_
```YAML
Language: Cpp
```
___
*[`build/repositories.bzl` line 58 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhQjVjCiD9UktAC_vb:-NNhQjVjCiD9UktAC_vc:b-b3k6x6) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/build/repositories.bzl#L58)):*
> ```Starlark
>         #        sha256 = "a30369e28ae3788356b734239559f3d0c035d9121963ab00a797615364d4f0c4",
>         #        version = "0.3.0",
>         sha256 = None,
> ```

Specify the sha256 even when using a commit.



_Code quote:_
```Starlark
None
```
___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 255 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhRCUy0FsyrNbRcZmS:-NNhRCUy0FsyrNbRcZmT:bryb74y) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L255)):*
> ```cpp
> // returns the number of such noise registers added.
> absl::StatusOr<int64_t> AddBlindedHistogramNoise(
>     ProtocolCryptor &protocol_cryptor, int total_sketches_count,
> ```

Don't change the existing pointer/reference style. Keep the trailing space style that we're already using.

See https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions



_Code quote:_
```cpp
ProtocolCryptor &protocol_crypto
```
___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 263 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhSO2tD7sbf3seijn7:-NNhSO2tD7sbf3seijn8:b-hlyrj1) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L263)):*
> ```cpp
> 
>   int64_t noise_register_added = 0;
>   math::DistributedRandomNoise *pDistributedRandomNoise;
> ```

We use `lower_snake_case` for local variables, and do not use any kind of "Hungarian"/type prefix notation for types on local variables.

See https://google.github.io/styleguide/cppguide.html#Variable_Names



_Suggestion:_
```cpp
distributed_random_noise
```
___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 299 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhTAa87_6QKjHHZtwC:-NNhTAa90HXbrDp4M2WY:bhva80m) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L299)):*
> ```cpp
>   }
> 
>   delete pDistributedRandomNoise;
> ```

Why are we using a pointer with heap allocation when it looks like the lifetime is restricted to the scope of this function?

File: src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc [hunk 1]
```diff
@@ -289,6 +295,8 @@ absl::StatusOr<int64_t> AddBlindedHistogramNoise(
       }
     }
   }
+
+  delete pDistributedRandomNoise;
```
Comment (wangyaopw) [inline src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc:299]: avoid using raw pointer, i.e. no `new/delete` in the code.

either just use an instance, or use smart pointers

**Review by wangyaopw — CHANGES_REQUESTED** [review]

**Review by kungfucraig — CHANGES_REQUESTED** [review]
DistributedGeometricRandomNoise is not a great name for a class that is a function object. DistributedNoiser would be more appropriate.

> Reviewed 2 of 6 files at r1, 1 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-:-NNiCjo97YUl0m69m8ZP:b-1xvy5m)* status: 3 of 6 files reviewed, 7 unresolved discussions (waiting on @iverson52000, @SanjayVas, and @uakyol)

___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 256 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNiCWb0DGKhMesGIES-:-NNiCWb0DGKhMesGIES0:b-v6heid) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/1913764b997e389780d09d82ef91749779bc98c0/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L256)):*
> ```cpp
> absl::StatusOr<int64_t> AddBlindedHistogramNoise(
>     ProtocolCryptor& protocol_cryptor, int total_sketches_count,
>     const math::DistributedGeometricRandomComponentOptions& options,
> ```

It seems to me that each of these functions that you are having take sigma should instead omit sigma and replace the GeoComponentOptions with an instance of the base DistRandomNoise class.

Then the client can decide what kind of noise to introduce and pass the right instance.

**Comment by iverson52000** [comment]
Make sense. I will update the class names in any-sketch

**Review by iverson52000 — COMMENTED** [review]
Done

> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-:-NOLYBlh1AaD3oVc3rDP:bb6un1j)* status: 1 of 7 files reviewed, 7 unresolved discussions (waiting on @kungfucraig, @SanjayVas, @stevenwarejones, @uakyol, and @wangyaopw)

___
*[`.clang-format` line 1 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhU_B13iT-m8bcrnOU:-NOLYDgJ5vJCUvd0Vjlx:bwppva0) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/.clang-format#L1)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

We use `--style=Google` for more than just C++, e.g. protobuf. See https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/main/docs/code-style.md#other-languages

Note that it looks like this PR incorrectly reformats some code in a way that does not match this style.
</blockquote></details>

Done. Thanks for the reminder.
___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 255 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhRCUy0FsyrNbRcZmS:-NOLYLgPBP4__9IzwXvu:bttqyzn) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L255)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

Don't change the existing pointer/reference style. Keep the trailing space style that we're already using.

See https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions
</blockquote></details>

My CLion had some funky behavior when formatting the code. Now should be fixed
___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 263 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhSO2tD7sbf3seijn7:-NOLYlbB-YcROQotkNag:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L263)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

We use `lower_snake_case` for local variables, and do not use any kind of "Hungarian"/type prefix notation for types on local variables.

See https://google.github.io/styleguide/cppguide.html#Variable_Names
</blockquote></details>

Done.
___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 299 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhTAa87_6QKjHHZtwC:-NOLYnq-032993ILb6Qa:b2ze8rv) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L299)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

Why are we using a pointer with heap allocation when it looks like the lifetime is restricted to the scope of this function?
</blockquote></details>

The main reason is I'll need to let the pointer points to different noiser instance later. Per Yao's suggestion, change to using smart pointers
___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 256 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNiCWb0DGKhMesGIES-:-NOLYgjl1m-_94T8vHlB:b-5b7t16) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/1913764b997e389780d09d82ef91749779bc98c0/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L256)):*
<details><summary><i>Previously, kungfucraig (Craig Wright) wrote…</i></summary><blockquote>

It seems to me that each of these functions that you are having take sigma should instead omit sigma and replace the GeoComponentOptions with an instance of the base DistRandomNoise class.

Then the client can decide what kind of noise to introduce and pass the right instance.
</blockquote></details>

Done. Good suggestion thanks!
___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 299 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNcW545Rgq0ScRhSjiC-r2-299:-NOLYn_OCn0sBa9vRPuG:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/1913764b997e389780d09d82ef91749779bc98c0/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L299)):*
<details><summary><i>Previously, wangyaopw (Yao Wang) wrote…</i></summary><blockquote>

avoid using raw pointer, i.e. no `new/delete` in the code.

either just use an instance, or use smart pointers
</blockquote></details>

Done.

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> Reviewed 1 of 4 files at r2, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: all files reviewed, 4 unresolved discussions (waiting on @iverson52000, @kungfucraig, @stevenwarejones, @uakyol, and @wangyaopw)

___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 299 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhTAa87_6QKjHHZtwC:-NOMgunEEoGZy8lEGnms:b-d4oqlj) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L299)):*
<details><summary><i>Previously, iverson52000 (Albert Hsu) wrote…</i></summary><blockquote>

The main reason is I'll need to let the pointer points to different noiser instance later. Per Yao's suggestion, change to using smart pointers
</blockquote></details>

I don't follow. I don't see the pointer being reassigned here, nor any reference to it being kept after this function completes. It really seems like this can be passed in as `math::DistributedNoiser&`

**Review by iverson52000 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: all files reviewed, 4 unresolved discussions (waiting on @kungfucraig, @SanjayVas, @stevenwarejones, @uakyol, and @wangyaopw)

___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 299 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhTAa87_6QKjHHZtwC:-NOQi4ppDsoqgSPkh4Eg:b-ol2w3f) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L299)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

I don't follow. I don't see the pointer being reassigned here, nor any reference to it being kept after this function completes. It really seems like this can be passed in as `math::DistributedNoiser&`
</blockquote></details>

I mean something like this. But this will be done in a separate PR



_Code snippet:_
```cpp
std::unique_ptr<math::DistributedNoiser> distributed_noiser;

if (use_gaussian_noiser) {
  distributed_noiser =
      std::make_unique<math::DistributedDiscreteGaussiandNoiser>(
          gaussian_options)
} else if (use_laplace_noiser) {
  distributed_noiser =
      std::make_unique<math::DistributedGeometricNoiser>(geometric_options)
}

AddBlindedHistogramNoise(protocol_cryptor, total_sketches_count, distributed_noiser, data)
```

**Review by iverson52000 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: all files reviewed, 4 unresolved discussions (waiting on @kungfucraig, @SanjayVas, @stevenwarejones, @uakyol, and @wangyaopw)

___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 299 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhTAa87_6QKjHHZtwC:-NORBbIg1clKhuBZvfql:b-7a4ns7) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L299)):*
<details><summary><i>Previously, wangyaopw (Yao Wang) wrote…</i></summary><blockquote>

`AddBlindedHistogramNoise` can take a `math::DistributedNoiser& distributed_noiser` as the parameter. 

Then I think it is OK to just define 

```
math::DistributedNoiser distributed_noiser = DistributedDiscreteGaussianNoiser(gaussian_options)
```

and use it as an instance instead of a pointer
</blockquote></details>

Yeah this will work too

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: all files reviewed, 4 unresolved discussions (waiting on @iverson52000, @kungfucraig, @stevenwarejones, @uakyol, and @wangyaopw)

___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 299 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhTAa87_6QKjHHZtwC:-NOW6DKV8HWqjXaP9yiN:b-43o0ot) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L299)):*
<details><summary><i>Previously, iverson52000 (Albert Hsu) wrote…</i></summary><blockquote>

Yeah this will work too
</blockquote></details>

To clarify, how you choose to instantiate the Noiser is independent of how you pass it to this function (`AddBlindedHistogramNoise`). This function doesn't care whether you're allocating it on the heap using unique_ptr. What matters is the lifecycle of the Noiser w.r.t. this function. Using `unique_ptr` as the parameter type means you're passing ownership of the pointer to this function. That's entirely unnecessary, as it's entirely fine for the Noiser's lifecycle to instead be tied to the scope of the calling function. Therefore, the Noiser should be passed to this function as a `const` reference.

**Review by iverson52000 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: all files reviewed, 4 unresolved discussions (waiting on @kungfucraig, @SanjayVas, @stevenwarejones, @uakyol, and @wangyaopw)

___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 299 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhTAa87_6QKjHHZtwC:-NOp75Yf1bnEXUITXX3M:b-tl2jd) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L299)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

To clarify, how you choose to instantiate the Noiser is independent of how you pass it to this function (`AddBlindedHistogramNoise`). This function doesn't care whether you're allocating it on the heap using unique_ptr. What matters is the lifecycle of the Noiser w.r.t. this function. Using `unique_ptr` as the parameter type means you're passing ownership of the pointer to this function. That's entirely unnecessary, as it's entirely fine for the Noiser's lifecycle to instead be tied to the scope of the calling function. Therefore, the Noiser should be passed to this function as a `const` reference.
</blockquote></details>

Thanks for the clarification Sanjay! I definitely need more experience with C++.

**Review by iverson52000 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: 5 of 7 files reviewed, 4 unresolved discussions (waiting on @kungfucraig, @SanjayVas, @stevenwarejones, @uakyol, and @wangyaopw)

___
*[`src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc` line 299 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhTAa87_6QKjHHZtwC:-NP-Nx7-5uKWkWDrzsaN:bb6un1j) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/src/main/cc/wfa/measurement/internal/duchy/protocol/liquid_legions_v2/liquid_legions_v2_encryption_utility.cc#L299)):*
<details><summary><i>Previously, iverson52000 (Albert Hsu) wrote…</i></summary><blockquote>

Thanks for the clarification Sanjay! I definitely need more experience with C++.
</blockquote></details>

Done

**Review by SanjayVas — COMMENTED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: all files reviewed, 3 unresolved discussions (waiting on @iverson52000, @kungfucraig, @stevenwarejones, @uakyol, and @wangyaopw)

**Review by iverson52000 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: 6 of 7 files reviewed, 3 unresolved discussions (waiting on @kungfucraig, @SanjayVas, @stevenwarejones, @uakyol, and @wangyaopw)

___
*[`build/repositories.bzl` line 58 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846#-NNhQjVjCiD9UktAC_vb:-NPOjrae9LABfjQEBUmq:b17jv4i) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/6958188604f24c95c15be5910174274094317e96/build/repositories.bzl#L58)):*
<details><summary><i>Previously, SanjayVas (Sanjay Vasandani) wrote…</i></summary><blockquote>

Specify the sha256 even when using a commit.
</blockquote></details>

Done. Updated the new release version

**Review by SanjayVas — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: all files reviewed, 2 unresolved discussions (waiting on @kungfucraig, @stevenwarejones, @uakyol, and @wangyaopw)

**Review by wangyaopw — COMMENTED** [review]
> Reviewed 1 of 4 files at r2, 4 of 6 files at r3, 1 of 2 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: all files reviewed, 2 unresolved discussions (waiting on @kungfucraig, @stevenwarejones, and @uakyol)

**Review by wangyaopw — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: all files reviewed, 1 unresolved discussion (waiting on @kungfucraig, @stevenwarejones, and @uakyol)

**Review by kungfucraig — APPROVED** [review]
> Reviewed 4 of 6 files at r3, 1 of 2 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @stevenwarejones and @uakyol)

**Review by iverson52000 — COMMENTED** [review]
> Reviewed 1 of 4 files at r2, 4 of 6 files at r3, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/846)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @stevenwarejones and @uakyol)

**Pull request merged**